commit e8c0722927e8fc112d78cc0435d336fcd7e3507c
Author: Vladimir Oltean <vladimir.oltean@nxp.com>
Date:   Fri Oct 1 18:15:27 2021 +0300

    net: mscc: ocelot: write full VLAN TCI in the injection header
    
    The VLAN TCI contains more than the VLAN ID, it also has the VLAN PCP
    and Drop Eligibility Indicator.
    
    If the ocelot driver is going to write the VLAN header inside the DSA
    tag, it could just as well write the entire TCI.
    
    Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/include/linux/dsa/ocelot.h b/include/linux/dsa/ocelot.h
index 435777a0073c..0fe101e8e190 100644
--- a/include/linux/dsa/ocelot.h
+++ b/include/linux/dsa/ocelot.h
@@ -210,9 +210,9 @@ static inline void ocelot_ifh_set_tag_type(void *injection, u64 tag_type)
 	packing(injection, &tag_type, 16, 16, OCELOT_TAG_LEN, PACK, 0);
 }
 
-static inline void ocelot_ifh_set_vid(void *injection, u64 vid)
+static inline void ocelot_ifh_set_vlan_tci(void *injection, u64 vlan_tci)
 {
-	packing(injection, &vid, 11, 0, OCELOT_TAG_LEN, PACK, 0);
+	packing(injection, &vlan_tci, 15, 0, OCELOT_TAG_LEN, PACK, 0);
 }
 
 #endif